libpod/runtime_volume_common.go

```diff
@@ -151,15 +151,33 @@ func (r *Runtime) newVolume(ctx context.Context, noCreatePluginVolume bool, opti
 		if err := os.MkdirAll(volPathRoot, 0700); err != nil {
 			return nil, fmt.Errorf("creating volume directory %q: %w", volPathRoot, err)
 		}
-		if err := idtools.SafeChown(volPathRoot, volume.config.UID, volume.config.GID); err != nil {
-			return nil, fmt.Errorf("chowning volume directory %q to %d:%d: %w", volPathRoot, volume.config.UID, volume.config.GID, err)
+		uid, gid := func() (int, int) {
+			uid := volume.uid()
+			gid := volume.gid()
+			switch {
+			case uid != -1 && gid != -1:
+				return uid, gid
+			case uid == -1:
+				return os.Getuid(), gid
+			case gid == -1:
+				return uid, os.Getgid()
+			}
```

Comment:
Personal nit, YMMV.
In general, I've not been a fan of equality tests for -1.  Can we convert these to `uid > -1` or `uid < 0` or some such?